### PR TITLE
Create hook useDataTransaction

### LINKED ISSUE #330: [FEAT] Create Hook: `useDataTransaction`
## Create Hook: `useDataTransaction`

Develop a Hook that provides relevant information obtained from `RpcProvider` from "starknet" regarding transactions. You can refer to the example in the image below as a starting point for [this file](https://github.com/Nadai2010/scaffold-stark-2/blob/d109f31176e7f0a7b8014e24fbdc0ce22ed7736f/packages/nextjs/hooks/scaffold-stark/useScaffoldTx.ts), should always detect the network.

image

## Proposed Solutions or Ideas

You may include information such as:

- Block status
- Block hash
- Starknet version
- Any other data that could be relevant (example TPS, UOPS, Total TX ...) similar explorer Starkscan or Voyager. 

image

image

### PR DESCRIPTION
**Pull request by cidkagenow**
Fixes #330 

add useDataTransaction

## Types of change

- [ ] Feature
- [ ] Enhancement

### CODE REVIEW

File: packages/nextjs/components/scaffold-stark/CustomConnectButton/index.tsx [hunk 1]
```diff
@@ -24,6 +24,8 @@ export const CustomConnectButton = () => {
   const [accountChainId, setAccountChainId] = useState<bigint>(0n);
   const { chain } = useNetwork();
 
+
+
   const blockExplorerAddressLink = useMemo(() => {
```
Comment (iossocket) [inline packages/nextjs/components/scaffold-stark/CustomConnectButton/index.tsx:29]: two empty lines for?

File: packages/nextjs/hooks/scaffold-stark/useDataTransaction.ts [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+  blockNewroot: string;
+  totalTransactions: number;
+  tps: number | null;
+  gasprice: string;
+  gaspricefri: string;
+  timeDiff: number;
+  averageFeeUSD: string;
+}
+
+export const useDataTransaction = (blockNumber: number) => {
```
Comment (iossocket) [inline packages/nextjs/hooks/scaffold-stark/useDataTransaction.ts:23]: pls add unit tests for this hook, thanks

**Review by Nadai2010 — CHANGES_REQUESTED** [review]
When testing I receive several isuue, please solve it.

Captura desde 2024-12-19 08-39-06

File: packages/nextjs/hooks/scaffold-stark/useDataTransaction.ts [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+import { RpcProvider } from "starknet";
+import { useTargetNetwork } from "./useTargetNetwork";
+
+interface BlockData {
+  classeslength: number;
+  transaction: number;
+  blockStatus: string;
+  blockNumber: number;
+  blockHash: string;
+  blockVersion: string;
```
Comment (Nadai2010) [inline packages/nextjs/hooks/scaffold-stark/useDataTransaction.ts:11]: I I think it is better to call it starknetVersion here, as it is really the version that is returned.

File: packages/nextjs/hooks/scaffold-stark/useDataTransaction.ts [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+  blockTimestamp: number;
+  blockTransactions: string[];
+  blockNewroot: string;
```
Comment (Nadai2010) [inline packages/nextjs/hooks/scaffold-stark/useDataTransaction.ts:14]: The blockNerRoot I think it is better to change it to parentBlockHash,

File: packages/nextjs/hooks/scaffold-stark/useDataTransaction.ts
Comment (Nadai2010) [inline packages/nextjs/hooks/scaffold-stark/useDataTransaction.ts:1]: A quick test using the Hook in Sepolia shows a good comparison in the metrics. (We should review the conversations of some data, that is already part of our hook. IMO) @metalboyrick 

imagen

imagen


In Mainnet Todo OK

imagen

imagen

In Devnet OK

imagen

**Comment by metalboyrick** [comment]
yep thanks @Nadai2010 @iossocket for the review, can resolve the comments ser @cidkagenow !

File: packages/nextjs/contracts/deployedContracts.ts [hunk 5]
```diff
@@ -3,6 +3,265 @@
  * You should not edit it manually or your changes might be overwritten.
  */
 
-const deployedContracts = {} as const;
+const deployedContracts = {
```
[the remark this answers is not in the document]
Comment (metalboyrick) [reply packages/nextjs/contracts/deployedContracts.ts:6]: hey @cidkagenow seems like this hasnt been resolved yet

File: packages/nextjs/hooks/scaffold-stark/useDataTransaction.ts [hunk 6]
```diff
@@ -75,7 +84,7 @@ export const useDataTransaction = (blockNumber: number) => {
             return data.ethereum.usd;
           } catch (error) {
             console.error("Error fetching ETH price:", error);
-            return 4000;
+            return 4000; // Valor por defecto en caso de error
```
Comment (Nadai2010) [inline packages/nextjs/hooks/scaffold-stark/useDataTransaction.ts:87]: Please remove or put the comment in English, the rest would be fine by me.

**Review by Nadai2010 — CHANGES_REQUESTED** [review]

**Review by metalboyrick — APPROVED** [review]
looks good to me!

**Review by Nadai2010 — APPROVED** [review]

**Pull request merged**